extensions/wikia/ImageReview/modules/PromoteImage/PromoteImageReviewTask.php

```diff
@@ -1,429 +1,429 @@
-
-		$wikiData = $this->model->getWikiData($sourceWikiId, $sourceWikiLang, $this->helper);
-		if(!empty($wikiData['images'])) {
-			$currentImages = $wikiData['images'];
-
-			foreach($currentImages as $imageName) {
-				$promoImage = PromoImage::fromPathname($imageName);
-				if( $promoImage->isAdditional() && !in_array($promoImage->getPathname(), $deletedImages) ) {
-					$data['city_images'][] = $promoImage->getPathname();
-				}
-			}
-		}
-
-		if( isset($data['city_images']) ) {
-			$data['city_images'] = json_encode($data['city_images']);
-		}
-
-		return $data;
-	}
-
-	/**
-	 * @desc Adding data of wikis imported on wikia.com by import script
-	 *
-	 * @param $targetWikiId
-	 * @param $targetWikiLang
-	 * @param $images
-	 */
-	protected function addImagesToPromoteDb($targetWikiId, $targetWikiLang, $images) {
-		global $wgExternalSharedDB;
-
-		foreach( $images as $image ) {
-			$imageData = new stdClass();
-
-			$promoImage = PromoImage::fromPathname($image['name'])->ensureCityIdIsSet($targetWikiId);
-
-			$imageData->city_id = $targetWikiId;
-			$imageData->page_id = $image['id'];
-			$imageData->city_lang_code = $targetWikiLang;
-			$imageData->image_index =  $promoImage->getType();
-			$imageData->image_name = $promoImage->getPathname();
-			$imageData->image_review_status = ImageReviewStatuses::STATE_APPROVED;
-			$imageData->last_edited = date('Y-m-d H:i:s');
-			$imageData->review_start = null;
-			$imageData->review_end = null;
-			$imageData->reviewer_id = null;
-
-			$imagesToAdd[] = $imageData;
-		}
-		$dbm = wfGetDB(DB_MASTER, array(), $wgExternalSharedDB);
-
-		$deleteArray = array();
-		$insertArray = array();
-		foreach( $imagesToAdd as $image ) {
-			$tmpArr = array();
-			foreach( $image as $field => $value ) {
-				$tmpArr[$field] = $value;
-			}
-			$insertArray[] = $tmpArr;
-			$deleteArray[] = $image->page_id;
-		}
-
-		$dbm->begin(__METHOD__);
-		$sql = 'DELETE FROM city_visualization_images WHERE page_id IN (' . $dbm->makeList($deleteArray) . ') AND city_id = \'' . $targetWikiId . '\'';
-
-		$dbm->query($sql);
-		$dbm->commit(__METHOD__);
-
-		$dbm->begin(__METHOD__);
-		$dbm->insert(
-			'city_visualization_images',
-			$insertArray
-		);
-		$dbm->commit(__METHOD__);
-	}
-
-	function getForm($title, $errors = false) {
-	}
-
-	function submitForm() {
-	}
-}
+<?php
+
+/**
+ * TaskManager task to go through a list of images and delete them.
+ */
+
+//use \Wikia\Logger\WikiaLogger; // does not autoload inside maintenance script
+ 
+
+class PromoteImageReviewTask extends BatchTask {
+	var $mType,
+		$mVisible,
+		$mArguments,
+		$mMode,
+		$mAdmin,
+		$records,
+		$title,
+		$namespace,
+		$mUser,
+		$mUsername;
+
+	protected $helper;
+	protected $model; /** @var CityVisualization $model */
+	protected $corporatePagesIds = array();
+	protected $dbNamesToBeSkipped = array();
+
+	function __construct($params = array()) {
+		$this->mType = 'promoteimagereview';
+		$this->mVisible = false; // do not show form for this task
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)